src/protocols/ranking.js

```diff
@@ -70,9 +70,9 @@ class Ranking {
 			return;
 		}
 
-		// Check if method is a MK7 patched method
-		if (packet.connection.accessKey === '6181dff1') {
-			RankingMK7.handlePacket(packet);
+		// Check if method is a legacy method
+		if (packet.connection.accessKey === '6181dff1' || packet.connection.accessKey === '0fabeff2') {
+			RankingLegacy.handlePacket(packet);
```

Comment:
Now that we know this is likely a rewritten protocol, and not game-specific patches, we should opt to check the protocol version rather than specific games

Both Mario Kart 7 and Mario Tennis Open using Ranking version 2.x.x. Mario Kart 7 is Ranking 2.4.3 and Mario Tennis Open is Ranking 2.6.1. It's very likely that this protocol was rewritten in version 3.0.0

Even if we don't want to assume that Ranking 2.x.x is always the legacy protocol, at the very least we know that 2.6.1is the current latest version using the legacy protocol. So anything using 2.6.1 or earlier should be consider to be using the legacy protocol